bn_mp_fwrite.c

```diff
@@ -19,10 +19,33 @@ mp_err mp_fwrite(const mp_int *a, int radix, FILE *stream)
       return MP_MEM;
    }
 
+   /*
+      Since mp_radix_sizeinbase() can overshoot by one (two with MP_8BIT)
```

Comment:
> The string is NUL terminated so everything is fine IMO.

It is more or less only a problem with `fwrite` which would write more than expected by the user. Not a problem if it writes to `stdout` or `stderr` but might result in a lot of cussing&cursing if used elsewhere.

> To get the correct length, why not change the length parameter of `mp_to_radix()` to a `size_t*`?

Yes, changing `mp_to_radix` would be the simplest way to do it because we already now the exact number at that time. The API has already been changed and not that long ago, so it wouldn't be a problem to do it again, I think.

OK, do it here or in another PR? (Somebody should send @MasterDuke17 a note which I hereby did. If Gihub's "you got mentioned" notification mechanism still works, that is).

Ah, all green. Squash or add the `mp_to_radix()` change?

Reply:
Hm, does this mean anything for my PR? It looks like my new `mp_to_decimal` already has its maxlen as `size_t`.

Reply:
@MasterDuke17 
> Hm, does this mean anything for my PR?

Yes, you would need to put the exact amount of digits plus the zero back in `maxlen`.

But wait until it's actually done.

> It looks like my new `mp_to_decimal` already has its maxlen as `size_t`.

You would need to change the prototype from
```c
mp_err mp_to_decimal(const mp_int *a, char *str, size_t maxlen);
```
to
```c
mp_err mp_to_decimal(const mp_int *a, char *str, size_t *maxlen);
```
(and change the rest according to that.)

So nothing big, I just thought it would be nice if you get a warning to avoid uncomfortable surprises and can prepare for it in advance.

Thank you for your patience with us!

Reply:
Ah, I see. Ok, I'll make those changes to my PR after this PR is merged.

Reply:
> OK, do it here or in another PR?

please in another PR

Reply:
> please in another PR

In #348 



> >   How about calling this `mp_radix_size_approx()`?
> or `mp_radix_size_fast()`?

The name `sizeinbase` is shamelessly stolen from GMP where it also shows the same behaviour and because programmers always have a hard time to come up with new and exiting names I took that instead.
Yeah, call me lazy, that's what I am ;-)